middleware/validate.js

```diff
@@ -0,0 +1,168 @@
+module.exports = {
+  validator: function (req, res, next) {
+
+    /************************
+     *  ARTICLES
+    ************************/
+    if (req.originalUrl.slice(0, 9) === '/articles') {
+      const title = req.params.title;
+      const articleDB = require('../db/articles.js');
+      const database = articleDB.getArticles().articles;
+
+      // GET '/:title'
+      if (req.method === 'GET' && req.url === `/${title}`) {
+        const params = req.params;
+        const articleIndex = articleDB.findArticle(params.title);
+
+        if (typeof articleIndex !== 'number') {
+          res.status(500);
+          return res.json({ "Error": "Couldn't find that product in our database." });
+        }
+      }
+
+      // GET '/:title/edit'
+      if (req.method === 'GET' && req.url === `/${title}/edit`) {
+        let params = req.params;
+        let articleIndex = articleDB.findArticle(params.title);
```

Comment:
Please create a function that can get an article by title and just use that here instead.